server/routes/addEditors.ts

```diff
@@ -106,47 +117,57 @@ const addEditors = async (models: DB, req: Request, res: Response, next: NextFun
 
   await thread.save();
 
-  if (collaborators?.length > 0) await Promise.all(collaborators.map(async (collaborator) => {
-    if (!collaborator.User) return; // some Addresses may be missing users, e.g. if the user removed the address
+  if (collaborators?.length > 0)
+    await Promise.all(
+      collaborators.map(async (collaborator) => {
+        if (!collaborator.User) return; // some Addresses may be missing users, e.g. if the user removed the address
 
-    await models.Subscription.emitNotifications(
-      models,
-      NotificationCategories.NewCollaboration,
-      `user-${collaborator.User.id}`,
-      {
-        created_at: new Date(),
-        root_id: +thread.id,
-        root_type: ProposalType.OffchainThread,
-        root_title: thread.title,
-        comment_text: thread.body,
-        chain_id: thread.chain,
-        author_address: author.address,
-        author_chain: author.chain,
-      },
-      {
-        user: author.address,
-        url: getProposalUrl('discussion', thread),
-        title: req.body.title,
-        bodyUrl: req.body.url,
-        chain: thread.chain,
-        body: thread.body,
-      },
-      req.wss,
-      [ author.address ]
+        await models.Subscription.emitNotifications(
+          models,
+          NotificationCategories.NewCollaboration,
+          `user-${collaborator.User.id}`,
+          {
+            created_at: new Date(),
+            root_id: +thread.id,
+            root_type: ProposalType.OffchainThread,
+            root_title: thread.title,
+            comment_text: thread.body,
+            chain_id: thread.chain,
+            author_address: author.address,
+            author_chain: author.chain,
+          },
+          {
+            user: author.address,
+            url: getProposalUrl('discussion', thread),
+            title: req.body.title,
+            bodyUrl: req.body.url,
+            chain: thread.chain,
+            body: thread.body,
+          },
+          req.wss,
+          [author.address]
+        );
+      })
     );
-  }));
 
-  const finalEditors = await models.Collaboration.findAll({
+  const finalCollaborations = await models.Collaboration.findAll({
```

Comment:
Linters making it tough to see which changes are real—main backend fixes are in lines 153-170 of this file (/addEditors), and 68-84 of /deleteEditors route.